hedera-node/src/main/java/com/hedera/services/txns/contract/ContractCallTransitionLogic.java

```diff
@@ -128,30 +138,60 @@ public void doStateTransitionOperation(
 
 		// --- Do the business logic ---
 		TransactionProcessingResult result;
+		final var isDirectTokenCall = isTokenAccount && properties.enableDirectHTSTokenCalls();
+
 		if (relayerId == null) {
-			result = evmTxProcessor.execute(
-					sender,
-					receiver.canonicalAddress(),
-					op.getGas(),
-					op.getAmount(),
-					callData,
-					txnCtx.consensusTime());
+			if (isDirectTokenCall) {
+				result = evmTxSimulator.execute(
+						sender,
+						op.getGas(),
+						op.getAmount(),
+						callData,
+						txnCtx.consensusTime(),
+						aliasManager.resolveForEvm(receiver.canonicalAddress()),
+						offeredGasPrice,
+						maxGasAllowanceInTinybars,
+						null,
+						targetId.num(),
+						entityAccess.worldLedgers());
```

Comment:
I think we can add a specific `executeEth()` method specifically for eth txns, as we have separate methods in `evmTxProcessor`. In this way we won't pass `relayer`, `offeredGasPrice` and `maxGasAllowance`. Since we pass here the `offeredGasPrice` and `maxGasAllowance` we are prone to have the side-effect of misleading the `EvmTxnProcessorSimulator` to make gas calculations based on ethereum logic while we have a regular hedera call.